x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1419,6 +1419,22 @@ static void validateForFips(Settings settings) {
                     + " ] setting."
             );
         }
+        final var cacheHashAlgoSettings = settings.filter(k -> k.endsWith("cache.hash_algo"));
+        cacheHashAlgoSettings.keySet().forEach((key) -> {
+            final var setting = cacheHashAlgoSettings.get(key);
+            assert setting != null;
+            final var hashAlgoName = setting.toLowerCase(Locale.ROOT);
+            if (hashAlgoName.equals("ssha256") == false
+                && hashAlgoName.equals("sha1") == false
+                && hashAlgoName.startsWith("pbkdf2") == false) {
+                logger.warn(
+                    "Only SSHA256, SHA1, or PBKDF2 are allowed for secure hashing in a FIPS 140 JVM. "
+                        + "Please change the [{}] setting from [{}] to an appropriate value.",
```

Comment:
I have a few comments on this message:
1. Since this is a warning, I'd tone down it a bit. So instead of saying ".. are allowed ...", I suggest we change it to "...  are recommended ..." so it does _not_ read as strong as other messages that would cause node start failure.
2. I don't think we should recommend SHA1. Strictly speaking we should not use unsalted hash for password like information. In practice, it's probably not too bad since cache is relatively short-lived (24 hours for API keys). But as a general message, it's better to avoid it.
3. I don't it's wise to use PBKDF2 for cache hashing because of performance. But it is not wrong. So I am OK with it.

Reply:
Also wondering if we should go with something like "in-memory credential hashing" instead of "secure hashing" if we kick out SHA1. Wydt?

Reply:
I am no FIPS expert but I don't think it explicitly recommends any password hashing algorithms. PBKDF2 is the "semi"-standard because it is a clear FIPS choice for key derivation which happens to work like a hash. So there is some wiggle room here. But there are published attacks for SHA-1. So I find it hard to recommend it for password hashing, even when it is in memory.

> Also wondering if we should go with something like "in-memory credential hashing" instead of "secure hashing" for the message
This is a good point. I think we can be more explicit here because people may even question ssha256 without more context.